odoo/models.py

```diff
@@ -4240,6 +4244,17 @@ def _load_records(self, data_list, update=False):
                 if data.get('xml_id') and not data['xml_id'].startswith(prefix):
                     _logger.warning("Creating record %s in module %s.", data['xml_id'], module)
 
+        if self.env.context.get('import_file'):
+            existing_modules = self.env['ir.module.module'].search([]).mapped('name')
```

Comment:
`self.env['ir.module.module']._installed()` is the one you need 😃

Reply:
Checking all known modules instead of the installed ones is deliberate: if you import a record with a prefix that refers to a module that's not installed, it'll get removed when you do install the module.